flink/src/main/java/org/apache/iceberg/flink/actions/MigrateAction.java

```diff
@@ -0,0 +1,374 @@
+      throw new RuntimeException(String.format("The source table %s not exists ! ", hiveSourceTableName));
+    }
+
+    List<FieldSchema> fieldSchemaList = hiveTable.getSd().getCols();
+    List<FieldSchema> partitionList = hiveTable.getPartitionKeys();
+    fieldSchemaList.addAll(partitionList);
+    Schema icebergSchema = HiveSchemaUtil.convert(fieldSchemaList);
+    PartitionSpec spec = toPartitionSpec(partitionList, icebergSchema);
+
+    String hiveFormat = hiveTable.getSd().getInputFormat();
+    FileFormat fileFormat;
+    switch (hiveFormat) {
+      case PATQUET_INPUT_FORMAT:
+        fileFormat = FileFormat.PARQUET;
+        break;
+
+      case ORC_INPUT_FORMAT:
+        fileFormat = FileFormat.ORC;
+        break;
+
+      case AVRO_INPUT_FORMAT:
+        fileFormat = FileFormat.AVRO;
+        break;
+
+      default:
+        throw new UnsupportedOperationException("Unsupported file format");
+    }
+
+    TableIdentifier identifier = TableIdentifier.of(toNamespace(baseNamespace), icebergTableName);
+    String hiveLocation = hiveTable.getSd().getLocation();
+
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(TableProperties.DEFAULT_FILE_FORMAT, fileFormat.name());
+    if (!baseNamespace.isEmpty()) {
+      properties.put(FlinkCatalogFactory.BASE_NAMESPACE, baseNamespace.toString());
+    }
+
+    Table icebergTable;
+    if (icebergCatalog instanceof HadoopCatalog) {
+      icebergTable = icebergCatalog.createTable(identifier, icebergSchema, spec, properties);
+    } else {
+      icebergTable = icebergCatalog.createTable(identifier, icebergSchema, spec, hiveLocation, properties);
+    }
+
+    String nameMapping =
+        PropertyUtil.propertyAsString(icebergTable.properties(), TableProperties.DEFAULT_NAME_MAPPING, null);
+    MetricsConfig metricsConfig = MetricsConfig.fromProperties(icebergTable.properties());
+
+    List<DataFile> files = null;
+    if (spec.isUnpartitioned()) {
+      MigrateMap migrateMap = new MigrateMap(spec, nameMapping, fileFormat, metricsConfig);
+      DataStream<PartitionAndLocation> dataStream =
+          env.fromElements(new PartitionAndLocation(hiveLocation, Maps.newHashMap()));
+      DataStream<List<DataFile>> ds = dataStream.map(migrateMap);
+      try {
+        files = Lists.newArrayList(ds.executeAndCollect("migrate table :" + icebergTable.name())).stream()
+            .flatMap(Collection::stream).collect(Collectors.toList());
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      try {
+        files = importPartitions(spec, tableSource, nameMapping, fileFormat, metricsConfig, icebergTable.name());
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    AppendFiles append = icebergTable.newAppend();
+    files.forEach(append::appendFile);
+    append.commit();
+    return files.size();
+  }
+
+  private Namespace toNamespace(Namespace namespace) {
+    if (namespace.isEmpty()) {
```

Comment:
one probably slightly simpler approach is to use List to avoid the if-else. First add the levels from `baseNamespace`. Then add the `icebergDbName`. Then `toArray`

Reply:
Actually, I found a simpler one-liner.
```
Namespace.of(ArrayUtils.concat(baseNamespace.levels(), new String[] {DATABASE}))
```